### DISCUSSION TITLE
Best way to do graphql calls

### DISCUSSION THREAD

**Discussion by ScottChapman** [octokit/octokit.js#1784] [Q&A] [answered]
So, I'm converting all my use of octokit/rest.js -> graphql and wondering the best approach.

Right now I am using a few plugins (including retry & throttling), creating an instance of `@octokit/rest` and using i's `request` parameter to make my requests to `POST /graphql`

I realize there is an octokit/graphql.js project, but it doesn't appear to support those plugins.

Thoughts on best approach? I suppose I'm not limited to those two options; I could always pick up a different graphql client...

**Comment by gr2m** [comment, marked as answer]
you can use `octokit.graphql` to send GraphQL queries, it's an instance of `octokit/graphql.js`

`octokit.graphql` was added in v17

**Reply by ScottChapman** [reply]
ah, cool. I'm on v16 still, so good reason to upgrade. I assume that shares the same request object?

**Reply by gr2m** [reply]
Yes.
